galley/pkg/runtime/projections/serviceentry/handler.go

```diff
@@ -271,30 +261,26 @@ func (h *Handler) doUpdate(name resource.FullName) {
 	h.updateVersion()
 }
 
-func (h *Handler) toMcpResource(name resource.FullName, service resource.Entry, endpoints *resource.Entry,
-	serviceEntry proto.Message) (*mcp.Resource, bool) {
-
-	body, err := types.MarshalAny(serviceEntry)
-	if err != nil {
-		scope.Errorf("error serializing proto from source e: %v:", serviceEntry)
+func (h *Handler) toMcpResource(service *resource.Entry, endpoints *resource.Entry) (*mcp.Resource, bool) {
+	meta := mcp.Metadata{}
+	se := networking.ServiceEntry{}
+	if err := h.converter.Convert(service, endpoints, &meta, &se); err != nil {
+		scope.Errorf("error converting to ServiceEntry: %v", err)
 		return nil, false
 	}
 
-	createTime, err := types.TimestampProto(service.Metadata.CreateTime)
+	// Set the version on the metadata.
+	meta.Version = h.versionString()
+
+	body, err := types.MarshalAny(&se)
 	if err != nil {
-		scope.Errorf("error parsing resource create_time for event metadata (%v): %v", service.Metadata, err)
+		scope.Errorf("error serializing proto from source e: %v:", se)
```

Comment:
Does this log message look right with a colon in the end?

Reply:
fixed